### PR TITLE
Add support for None values in netFlow, value, and consumption fields

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
This pull request adds support for None values in the netFlow, value, and consumption fields of the Exchange, TotalProduction, and TotalConsumption classes. This allows for more flexibility in data input and better handling of missing data. The _none_safe_round function has also been added to safely round values that may be None.

This is more of a personal preference rather than hard requirement so I'm open to discussion about this. However this unifies how we handle values passed in append (both StorageMix and ProductionMix can already be None) and moves more logic to the classes rather than requirering it in the parsers.

If we want to go forward with it I'll create and update the tests for it.

### CODE REVIEW

File: electricitymap/contrib/lib/models/events.py [hunk 1]
```diff
@@ -382,10 +400,12 @@ def to_dict(self) -> dict[str, Any]:
 class TotalProduction(Event):
```
Comment (VIKTORVAV99) [inline electricitymap/contrib/lib/models/events.py:400]: We should also consider renaming this to TotalGeneration IMO so it's less likely to be confused with the production breakdown.

**Comment by mathilde-daugy** [comment]
This makes good sense to me. I think if @unitrium has no objection we can merge this np :)

**Review by mathilde-daugy — APPROVED** [review]

**Comment by VIKTORVAV99** [comment]
Thanks, if @unitrium don't have any objections I'll create some new tests for this and update tests if needed.

(as well as fixing the edge case were some parsers returned strings that was somehow converted automatically before (unwanted behaviour))

**Review by unitrium — APPROVED** [review]
No problems for me!
